### PR TITLE
feat(ui-watt): drawer content switch

### PR DESCRIPTION
**Pull request by ManBearTM**
## Description
This PR adds support for closing the drawer by "clicking outside", while still preventing it from closing if the click action triggers a call to `drawer.open()` (this behavior is identical to the drawer found in zenhub for instance). 

I also added a built-in loading state to the drawer for ease-of-use along with a story showing how to structure your code if you wish to reuse the same drawer component.

### CODE REVIEW

**Comment by dzhavat** [comment]
praise: the implementation works like a charm 👍 

suggestion: can we add a test that verifies the behaviour implemented in this PR?

File: libs/ui-watt/src/lib/components/drawer/watt-drawer.component.ts [hunk 1]
```diff
@@ -53,19 +57,30 @@ export class WattDrawerComponent implements AfterViewInit, OnDestroy {
    */
   opened = false;
 
+  bypassClickCheck = false;
```
[the remark this answers is not in the document]
Comment (ManBearTM) [reply libs/ui-watt/src/lib/components/drawer/watt-drawer.component.ts:61]: ff7ca5665b79990dfdaf75e379559e32ae307af3

**Comment by ManBearTM** [comment]
> praise: the implementation works like a charm 👍
> 
> suggestion: can we add a test that verifies the behaviour implemented in this PR?

@dzhavat I gave it a shot, but I couldn't really get it to work - seems pretty tricky to test this "click outside" behavior without tightly coupling the test to the implementation. If you have any ideas I'm all ears 🙂

**Comment by dzhavat** [comment]
> @dzhavat I gave it a shot, but I couldn't really get it to work - seems pretty tricky to test this "click outside" behavior without tightly coupling the test to the implementation. If you have any ideas I'm all ears 🙂

Hmmm ... the "closes drawer when another drawer is opened" test already does something in that direction. I don't know how a new test will be much different (from a user's perspective).

But maybe a test based on the `Loading` story where `closedOutput` is asserted for not being called when button 1 and 2 are clicked, and another assertion for the content of the drawer.

And a separate test for the `loading` input - that an element with class `watt-drawer__spinner` is present.

What do you think about such tests?

**Review by dzhavat — DISMISSED** [review]

**Review by dzhavat — APPROVED** [review]

**Pull request merged**